### PR TITLE
MC-1268 Adding operations for configuration reload & update

### PR DESCRIPTION
**Pull request by erosb**
Fixes [MC-1268](https://hazelcast.atlassian.net/browse/MC-1268)

Adds `reloadConfig` and `memberConfig` operations. These are remote interfaces for the `ConfigurationService#update()` methods implemented as part of [HZ-760](https://hazelcast.atlassian.net/browse/HZ-760).

OS PR: https://github.com/hazelcast/hazelcast/pull/20529
EE PR: https://github.com/hazelcast/hazelcast-enterprise/pull/4621

### CODE REVIEW

File: protocol-definitions/MC.yaml [hunk 1]
```diff
@@ -859,3 +859,37 @@ methods:
...
+      params:
+        - name: configUpdateResult
+          type: ConfigUpdateResult
+          since: 2.4
+          nullable: false
+          doc: the result of the configuration update
+  - id: 35
+    name: updateConfig
+    since: 2.4
+    doc: Reloads the configuration on the member from its config file, applies (merges) the new config to the current config
```
Comment (ramizdundar) [inline protocol-definitions/MC.yaml:879]: minor: Reloads -> Updates

File: protocol-definitions/custom/Custom.yaml [hunk 2]
```diff
@@ -996,3 +996,27 @@ customTypes:
...
+    returnWithFactory: true
+    params:
+      - name: configSectionName
+        type: String
+        since: 2.4
+        nullable: false
+      - name: configName
+        type: String
+        since: 2.4
+        nullable: false
```
Comment (ramizdundar) [inline protocol-definitions/custom/Custom.yaml:1022]: Name can be null.

File: protocol-definitions/MC.yaml [hunk 3]
```diff
@@ -859,3 +859,37 @@ methods:
...
          doc: Name of the Queue for which the age-related statistics should be reset to 0
     response: {}
+  - id: 34
+    name: reloadConfig
+    since: 2.4
+    doc: Reloads the configuration on the member from its config file, applies (merges) the new config to the current config
+    request:
+      retryable: false
+      partitionIdentifier: -1
+    response:
```
Comment (promanenko) [inline protocol-definitions/MC.yaml:869]: do we need to return a result of the update here?
We are going to get status events anyway, may be it makes sense to have this process async

File: protocol-definitions/MC.yaml [hunk 4]
```diff
@@ -859,3 +859,37 @@ methods:
...
+      params:
+        - name: configUpdateResult
```
Comment (promanenko) [inline protocol-definitions/MC.yaml:871]: we need to get information about the error in case of configuration update failure

Comment (erosb) [reply protocol-definitions/MC.yaml:871]: If we make this async (as per your other comment), then the failure reason will be included in the `ConfigUpdateFailedEvent` we receive on the MC side, won't it?

Comment (promanenko) [reply protocol-definitions/MC.yaml:871]: right

Comment (erosb) [reply protocol-definitions/MC.yaml:871]: ok thanks I will rework the PRs to make this operation async

**Review by mdumandag — APPROVED** [review]
Looks OK from the protocol's point of view.

We can merge this once the Hazelcast PRs are merged

**Review by promanenko — APPROVED** [review]

**Pull request merged**